CMakeLists.txt

```diff
@@ -49,6 +49,41 @@ if(NOT (ARM_TARGET MATCHES "^STM32F4") AND BUILD_WITH_LIBOPENCM3)
   message(FATAL_ERROR "Only STM32F4 series devices are supported via libopencm3")
 endif()
 
+option(USE_UART1 "Initializes UART1" OFF)
+option(USE_UART2 "Initializes UART2" OFF)
+
+# TODO: replace this by target_compile_definitions when utils get own target
+if(USE_UART1)
+  add_compile_definitions(USE_UART1)
+endif()
+
+if(USE_UART2)
+  add_compile_definitions(USE_UART2)
+endif()
```

Comment:
Can we move this closer to the source files which consume the compile definitions? Probably `samples/CMakeLists.txt` as long as the utils aren't own targets.